python/cudf/dataframe/dataframe.py

```diff
@@ -574,6 +574,12 @@ def __abs__(self):
     def __iter__(self):
         return iter(self.columns)
 
+    def equals(self, other):
+        for col in self.columns:
+            if not self[col].equals(other[col]):
+                return False
+        return True
```

Comment:
If col is not in `other[col]` this will return an error instead of False. Also this should probably check that the Indexes are equal as well.